### PR TITLE
Create records crud

### PR DESCRIPTION
**Pull request by pedroCervi**
:alien:

### CODE REVIEW

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class RecordsController < ApplicationController
+  before_action :authenticate_user!
```
Comment (cuzik) [inline app/controllers/records_controller.rb]: This should be moved to ApplicationController

File: app/controllers/records_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  def index
+    @records = current_user.records.all
```
Comment (cuzik) [inline app/controllers/records_controller.rb]:
```suggestion
    @records = current_user.records
```

File: app/controllers/records_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class RecordsController < ApplicationController
+  before_action :authenticate_user!
+
```
Comment (cuzik) [inline app/controllers/records_controller.rb:3]:
```suggestion
before_action :set_record only: [:show, :edit, :update, :destroy, :network_manage]

```

File: app/views/records/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<%= render 'form', record: @record %>
```
Comment (cuzik) [inline app/views/records/edit.html.erb]: We use on partial `_form` `@record` so we don't need pass local variables 

```suggestion
<%= render 'form' %>
```

File: app/views/records/new.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<%= render 'form', record: @record %>
```
Comment (cuzik) [inline app/views/records/new.html.erb]:
```suggestion
<%= render 'form' %>
```

File: app/views/records/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<div class="card mb-3">
+  <%= form_with(model: record, local: true) do |form| %>
```
Comment (cuzik) [inline app/views/records/_form.html.erb]: `record` is used here but all others call is `@record`

File: app/views/records/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    <div class="card-body">
+      <div class="row">
+        <div class="col-12">
+          <% if @record.errors.any? %>
+            <div id="error_explanation">
+              <h2>
+                <%= pluralize(@record.errors.count, "error") %> prohibited
+                this record from being saved:
```
Comment (cuzik) [inline app/views/records/_form.html.erb:10]: can we usar t('.error_massage') or only "impediram se ser salvo esse registro"

File: app/views/records/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+          <% end %>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</div>
+
+<br>
+
+<%= button_to "New Record", new_record_path, :method => "get", class: "btn btn-primary btn-lg btn-block login-button" %>
```
Comment (cuzik) [inline app/views/records/index.html.erb]:
```suggestion
<%= link_to(t('.new'), new_record_path, class: "btn btn-success float-right") %>
```

File: app/views/records/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    <div class="table-responsive">
+      <table class="table table-bordered" id="dataTable" width="100%" cellspacing="0">
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          <% @records.each do |record| %>
```
Comment (cuzik) [inline app/views/records/index.html.erb]: We usually use a partial to render record collections

```suggestion
          <% render (@records) %>
```

File: db/migrate/20190504135710_add_user_id_to_records.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToRecords < ActiveRecord::Migration[5.2]
+  def change
+    add_column :records, :user_id, :integer
```
Comment (cuzik) [inline db/migrate/20190504135710_add_user_id_to_records.rb]: Rails 5+ has belongs_to type like

```suggestion
    change_table :records do |t|
      t.belongs_to :user, index: true, null: false
    end
```

File: db/schema.rb [hunk 11]
```diff
@@ -10,10 +10,25 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_04_20_181834) do
+ActiveRecord::Schema.define(version: 2019_05_04_135710) do
 
   # These are extensions that must be enabled in order to support this database
+  enable_extension "citext"
```
Comment (cuzik) [inline db/schema.rb]: why?

File: db/schema.rb [hunk 12]
```diff
@@ -10,10 +10,25 @@
...
   enable_extension "plpgsql"
+  enable_extension "uuid-ossp"
```
Comment (cuzik) [inline db/schema.rb]: why?

File: spec/factories/records.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :record do
+    name { Faker::Movies::HitchhikersGuideToTheGalaxy.character }
+    description { Faker::Movies::HitchhikersGuideToTheGalaxy.quote }
+    value { Random.new.rand(0.0..9.9) }
+  end
```
Comment (cuzik) [inline spec/factories/records.rb:6]:
```suggestion
 
    user
  end
```

File: spec/controllers/records_controller_spec.rb [hunk 14]
```diff
@@ -0,0 +1,131 @@
+require 'rails_helper'
+
+RSpec.describe RecordsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:record) { create(:record, user: user) }
```
Comment (cuzik) [inline spec/controllers/records_controller_spec.rb:5]: We need `valid_attributes` and `record`?

File: app/controllers/records_controller.rb [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+  def show
+    render :show, locals: { record: record }
+  end
+
+  def new
+    @record = Record.new
+  end
+
+  def edit
+    render :edit, locals: { record: record }
```
Comment (cuzik) [inline app/controllers/records_controller.rb]: We use @record, so we can put in the `before_action` 

```
before_action :set_record, only: [:show, :edit, :update, :destroy]
```

```suggestion
```

File: app/controllers/records_controller.rb [hunk 16]
```diff
@@ -0,0 +1,58 @@
+class RecordsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @records = current_user.records.all
+  end
+
+  def show
+    render :show, locals: { record: record }
```
Comment (cuzik) [inline app/controllers/records_controller.rb]: We need locals? We use in show the local `record` but we have `@record` too

```suggestion
```

**Review by cuzik — CHANGES_REQUESTED** [review]

File: app/controllers/records_controller.rb (hunk 16 above)
Comment (pedroCervi) [reply app/controllers/records_controller.rb]: I think the use of variable instances inside views was a smell. But I'm okay with your suggestion

File: app/views/records/_form.html.erb (hunk 6 above)
Comment (pedroCervi) [reply app/views/records/_form.html.erb]: I meant to use locals everywhere but I'm changing accordingly to the `before_action` approach.

File: app/views/records/index.html.erb (hunk 9 above)
Comment (pedroCervi) [reply app/views/records/index.html.erb]: Good point.

File: app/views/records/index.html.erb (hunk 8 above)
Comment (pedroCervi) [reply app/views/records/index.html.erb]: I'm okay with using I18n more extensively but maybe we should focus on more basic things. We can, however, set the application correctly but don't enforce the use of i18n through the course. WDYT?

File: db/schema.rb (hunk 11 above)
Comment (pedroCervi) [reply db/schema.rb]: don't know,.

File: db/schema.rb (hunk 12 above)
Comment (pedroCervi) [reply db/schema.rb]: don't know also

File: spec/controllers/records_controller_spec.rb (hunk 14 above)
Comment (pedroCervi) [reply spec/controllers/records_controller_spec.rb:5]: Yes, we do. `record` is used for the show, edit and update endpoints, `valid_attributes` is used on the create action for example.

File: spec/factories/records.rb (hunk 13 above)
Comment (pedroCervi) [reply spec/factories/records.rb:6]: ? Do you mean to create a user relationship with this?

File: app/views/records/_form.html.erb (hunk 7 above)
Comment (pedroCervi) [reply app/views/records/_form.html.erb:10]: Same comment about i18n usage. I'm keeping the way it is for now. Let's change this later if we agree that i18n should be addressed on the course. WDYT?

File: spec/controllers/records_controller_spec.rb (hunk 14 above)
Comment (cuzik) [reply spec/controllers/records_controller_spec.rb:5]: Saquei

File: app/views/records/_form.html.erb (hunk 7 above)
Comment (cuzik) [reply app/views/records/_form.html.erb:10]: Agree >_<

File: app/views/records/index.html.erb (hunk 8 above)
Comment (cuzik) [reply app/views/records/index.html.erb]: Aham mas o resto acho que fica mais claro mesmo ser usar o `:method => "get"` não?!

File: spec/factories/records.rb (hunk 13 above)
Comment (cuzik) [reply spec/factories/records.rb:6]: É que tem um `belongs_to` pra `user` dentro de `record` não??

Comment (cuzik) [reply spec/factories/records.rb:6]: Ai quando tu for criar um `record` pela factory não precisa passar um `user` mas se passar ele usa o que tu passo e não cria outro dai.

Comment (pedroCervi) [reply spec/factories/records.rb:6]: Certo mano.

**Review by cuzik — APPROVED** [review]
:alien:

**Pull request merged**